processing/src/main/java/io/druid/query/dimension/DefaultDimensionSpec.java

```diff
@@ -64,17 +64,20 @@ public DimensionSpec apply(String input)
   }
 
   private static final byte CACHE_TYPE_ID = 0x0;
+  private final String dataSourceName;
   private final String dimension;
   private final String outputName;
   private final ValueType outputType;
 
   @JsonCreator
   public DefaultDimensionSpec(
+      @JsonProperty("dataSource") String dataSourceName,
```

Comment:
```dataSource``` is optional and can be omitted for single source queries. It must be specified for only multi source queries.